src/main/java/com/hms_networks/americas/sc/extensions/util/RawNumberValueUtils.java

```diff
@@ -0,0 +1,313 @@
+              + value2.getClass().getName());
+    }
+
+    // Return max value
+    return maxValue;
+  }
+
+  /**
+   * Get and return the sum of the specified values.
+   *
+   * @param values values to sum
+   * @return sum of values
+   * @throws IllegalArgumentException if values are not of the same type
+   * @since 1.15.7
+   */
+  public static Object getValueAvg(Object[] values) {
+    // Get sum of values
+    Object sum = getValueSum(values);
+
+    // Get average of values
+    Object avg = getValueDivide(sum, values.length);
+
+    // Return average
+    return avg;
+  }
+
+  /**
+   * Get and return the sum of the specified values.
+   *
+   * @param values values to sum
+   * @return sum of values
+   * @throws IllegalArgumentException if values are not of the same type
+   * @since 1.15.7
+   */
+  public static Object getValueSum(Object[] values) {
+    // Sum values by type
+    Object sum;
+    Object firstValue = values[0];
+    if (firstValue instanceof Integer) {
+      // Sum values
+      int sumInt = 0;
+      for (int i = 0; i < values.length; i++) {
+        sumInt += ((Integer) values[i]).intValue();
+      }
+      sum = new Integer(sumInt);
+    } else if (firstValue instanceof Long) {
+      // Sum values
+      long sumLong = 0;
+      for (int i = 0; i < values.length; i++) {
+        sumLong += ((Long) values[i]).longValue();
+      }
+      sum = new Long(sumLong);
+    } else if (firstValue instanceof Float) {
+      // Sum values
+      float sumFloat = 0;
+      for (int i = 0; i < values.length; i++) {
+        sumFloat += ((Float) values[i]).floatValue();
+      }
+      sum = new Float(sumFloat);
+    } else if (firstValue instanceof Double) {
+      // Sum values
+      double sumDouble = 0;
+      for (int i = 0; i < values.length; i++) {
+        sumDouble += ((Double) values[i]).doubleValue();
+      }
+      sum = new Double(sumDouble);
+    } else if (firstValue instanceof Short) {
+      // Sum values
+      short sumShort = 0;
+      for (int i = 0; i < values.length; i++) {
+        sumShort += ((Short) values[i]).shortValue();
+      }
+      sum = new Short(sumShort);
+    } else if (firstValue instanceof Byte) {
+      // Sum values
+      byte sumByte = 0;
+      for (int i = 0; i < values.length; i++) {
+        sumByte += ((Byte) values[i]).byteValue();
+      }
+      sum = new Byte(sumByte);
+    } else {
+      // Throw exception
+      throw new IllegalArgumentException(
+          "Cannot sum values of type " + values[0].getClass().getName());
+    }
+
+    // Return sum
+    return sum;
+  }
+
+  /**
+   * Get and return the value of the specified value divided by the specified divisor.
+   *
+   * @param value the value to divide
+   * @param divisor the divisor
+   * @return the value divided by the divisor
+   * @throws IllegalArgumentException if the value is not of a supported type
+   * @since 1.15.7
+   */
+  public static Object getValueDivide(Object value, int divisor) {
```

Comment:
divisor is an int, so this is meant for average calculation, not really general purpose. Should be private? Or should  divisor be object type?

Reply:
Updated so `divisor` is an `Object`.